Network Concept added (#30)
Human-Connection/Nitro-Documentation: concept/concept.md

# Network Concept

Human Connection is a nonprofit social, action and knowledge network that connects information to action and promotes positive local and global change in all areas of life. 

- **Social**: Interact with other people not just by commenting their posts, but by providing **Pro & Contra** arguments, give a **Versus** or ask them by integrated **Chat** or **Let's Talk**

- **Knowledge**: Read articles about interesting topics and find related posts in the **More Info** tab or by **Filtering** based on **Categories** and **Tagging** or by using the **Fulltext Search**.

- **Action**: Don't just read about how to make the world a better place, but come into **Action** by following provided suggestions on the **Action** tab provided by other people or **Organisations**.

## Features

The following features will be implemented. This gets done in three steps: 

> 1. First we will implement a basic feature set and provide an test system for to test the basic network functionality. 

Reviewer: ```suggestion
1. First we will implement a basic feature set and provide a test system to test the basic network functionality. 
```

> 2. In a second step we will go live into the public with an advanced feature set which includes also all the functionality in code and organisation that will be needed to drive a bigger public social network.

Reviewer: ```suggestion
2. In a second step we will make our prototype publicly available with an advanced feature set including the technology and organizational structure to drive a bigger public social network.
```

> 3. In a third step all the remaining Features will be implemented to build the full product. 

Reviewer: ```suggestion
3. In a third step all the remaining features will be implemented to build the full product. 
```
Will there ever be a "finished" product? :thinking: :wink:

Author: Corrected in updated branch network-concept

- Upload and Change Profile Picture
- Edit Social Media Accounts
- Edit Locale information
- Show and delete Bookmarks (later)
- Show Posts of a specific User
- Show Comments of a specific User

### Cockpit

[Clickdummy](https://preview.uxpin.com/24a2ab8adcd84f9a763d87ed27251351225e0ecd#/pages/99768919/simulate/sitemap?mode=i)

- Show Link to own Profile
- Show Friends Widget
- Show Favorites Widget
- Show Get Friends Widget
- Show popular Hashtags Widget
- Show Mini-Statistics Widget (all time)
- Show Chatrooms Widget
- Show List of Let's Talk requests with online status of requesting people

### Posts

- Creating Posts
- Upload Teaser Picture for Post
- Upload additional Pictures
- Editing Title and Content
- Allow embedded Conten (Videos, Sound, ...)
- Choosing a Category
- Adding Tags
- Choosing Language (e.g. German / English / French)
- Choosing Visibility (Public / Friends / Private)
- Shout Button for Posts
- Bookmark Posts (later)
- Optionally provide Let's Talk Feature
- Optionally provide Commenting Feature

### Comments

- Creating Comments 
- Deleting Comments
- Editing Comments
- Upvote comments of others

### Article List

- Show Posts by Tiles
- Show Posts as List

> - Filter by Category (Health and Wellbeing, Global Peace & Non-Violance, ...) 

Reviewer: ```suggestion
- Filter by Category (Health and Wellbeing, Global Peace & Non-Violence, ...) 
```

- Filter by Mood (Funny, Happy, Surprised, Cry, Angry, ...)
- Filter by Source (Connections, Following, Individuals, Non-Profits, ...)
- Filter by Posts & Tools (Post, Events, CanDos, ...)
- Filter by Format Type (Text, Pictures, Video, ...)
- Extended Filter (Continent, Country, Language, ...) 
- Sort Posts by Date
- Sort Posts by Shouts
- Sort Posts by most discussed

> - Sort Posts by most Emojis

Reviewer 2: Most different emojis or voting count in general?

Author: Clarified and changed in pushed in branch network-concept

### Blacklist

[Video](https://www.youtube.com/watch?v=-uDvvmN8hLQ)

- Blacklist Users

> - Blacklist Subjects

Reviewer 2: What is a subject in that case?

Author: Clarified and changed in updated branch network-concept

- Blacklist Tags

> - Switch on/off Adult Content

Reviewer: Mastodon and Peertube classify content (e.g. `nsfw` == not safe for work). Classification is a feature on its own.

Author: Yes, indeed.

### Search

- Search for Categories
- Search for Tags
- Fulltext Search

### CanDos

- Creating CanDos
- Editing Title and Content
- Choosing a Category
- Adding Tags
- Choosing Language (e.g. German / English / French)
- Choosing Visibility (Public / Friends / Private)
- Choosing Difficulty
- Editing Why - why should you do this
- Editing Usefulness - what is it good for

### Versus (interaction on existing Post)

- Create / edit / delete Versus

### Jobs

> - Create, edit and delete Jobs

Reviewer 2: Jobs can be created by everyone?

Author: Yes. Clarified and changed in updated branch network-concept

### Projects

- Create, edit and delete Projects
- Edit Title and Description for the Project
- Set Project Type
- Set and Edit Timeline for the Project
- Add Media to the Project
- Chat about the Project

### Pro & Contra

- Create Pro and Con (2-row)
- Add Arguments on Pro or Con Side
- Rate up Arguments
- Add Tags
- Attach Media

### Votes

- Create Votes (Surveys)
- Add Title and Description
- Let Users vote

> - Set Vote Type (Yes/No or Poll)

Reviewer 2: Why a type? Just create at least two options and name then yes no or do how ever you wish, type is not needed from my perspective.

Author: Agreed. Will change this and re-commit/push to branch

Author: Ripped out vote types after clarification with Dennis. See in updated branch network-concept

- Add Tags

### Bestlist

- Create Bestlist
- Create Votes (Surveys)
- Add Title and Description
- Add Tags
- Let Users vote for Best Item
- Set Settings (allow Uploads, allow Links, ...)

### Events

- Create Events

> - Add Title and Description

Reviewer 2: Joe about date and location? 😅

Author: did add these in updated branch network-concept

- Add Tags

### More Info

Shows autmatically releated information for existing post.

- Show related Posts
- Show Pros and Cons
- Show Bestlist
- Show Votes

> - Link to responding Chatroom 

Reviewer 2: Responding or corresponding?

Author: The latter. See updated branch network-concept
